src/Compilers/Core/VBCSCompiler/NamedPipeClientConnection.cs

```diff
@@ -19,6 +19,9 @@ internal sealed class NamedPipeClientConnection : IClientConnection
         // This is a value used for logging only, do not depend on this value
         private readonly string _loggingIdentifier;
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", 
+            MessageId = "System.Runtime.InteropServices.DangerousGetHandle", 
+            Justification = "We are using DangerousGetHandle for logging only.")]
         internal NamedPipeClientConnection(NamedPipeServerStream pipeStream)
```

Comment:
Looks like we just use a random number as identifier if the call to DangerousGetHandle() fails for whatever reason. Couldn't we just always use random number / GUID for the identifier (especially if we have noticed that DangerousGetHandle() does crash once in a while as the catch clause indicates)?